Framework/DataObjects/src/EventList.cpp

```diff
@@ -3664,11 +3664,7 @@ void EventList::filterByPulseTimeHelper(std::vector<T> &events, DateAndTime star
   while ((itev != itev_end) && (itev->m_pulsetime < start))
     itev++;
 
-  while ((itev != itev_end) && (itev->m_pulsetime < stop)) {
-    // Add the copy to the output
-    output.emplace_back(*itev);
-    ++itev;
-  }
+  std::copy_if(itev, itev_end, std::back_inserter(output), [stop](T &t) { return t.m_pulsetime < stop; });
```

Comment:
Wouldn't this be easier if we do
```c++
std::copy_if(
  events.begin(),
  events.end(),
  std::back_inserter(output),
  [start, stop] (T &t) {
    return (t.m_pulsetime >= start) && (t.m_pulsetime < stop);
  }
);
```

Reply:
Yes, I was just confused as to why it was split into two different loops in the original implementation. Looking back at the previous code I believe it was that way just to improve performance.